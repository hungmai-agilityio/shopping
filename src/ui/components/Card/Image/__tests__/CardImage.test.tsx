import { render, screen } from '@testing-library/react';

// Components
import CardImage from '@/ui/components/Card/Image';

// Mock next/image
jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => <img {...props} />
}));

describe('Card Image component', () => {
  const defaultProps = {
    src: 'https://i.ibb.co/52YmKP4/jean-3-1.webp',
    alt: 'Jean Image',
    width: 590,
    height: 350
  };

  it('renders Card image with default props', () => {
    render(<CardImage {...defaultProps} />);

    const card = screen.getByAltText('Jean Image');
    expect(card).toBeInTheDocument();
  });

  it('renders Card image with title', () => {
    render(<CardImage {...defaultProps} title="Title jean" />);

    const titleElement = screen.getByText('Title jean');
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveClass('bg-white text-purple-500');
  });

  it('renders Card image with blur effect when isBlur is true', () => {
    render(<CardImage {...defaultProps} isBlur={true} />);

    const card = screen.getByAltText('Jean Image');
    expect(card).toBeInTheDocument();
    expect(card).toHaveAttribute('placeholder', 'blur');
  });

  it('renders Card image without blur effect when isBlur is false', () => {
    render(<CardImage {...defaultProps} isBlur={false} />);

    const card = screen.getByAltText('Jean Image');
    expect(card).toBeInTheDocument();
    expect(card).toHaveAttribute('placeholder', 'empty');
  });

  it('should be render match to snapshot', () => {
    const { container } = render(
      <CardImage {...defaultProps} title="Title jean" />
    );

    expect(container).toMatchSnapshot();
  });
});
