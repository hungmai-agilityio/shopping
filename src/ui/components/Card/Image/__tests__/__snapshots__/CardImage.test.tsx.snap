// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Image component should be render match to snapshot 1`] = `
<div>
  <div
    class="relative inline-flex items-center justify-center"
  >
    <img
      alt="Jean Image"
      height="350"
      placeholder="empty"
      src="https://i.ibb.co/52YmKP4/jean-3-1.webp"
      style="object-fit: cover;"
      width="590"
    />
    <div
      class="absolute bg-white h-14 w-11/12 text-2xl font-bold text-purple-500 right-0 p-2 overflow-hidden truncate bottom-10"
    >
      Title jean
    </div>
  </div>
</div>
`;
